src/prefect/runner/submit.py

```diff
@@ -0,0 +1,196 @@
+import asyncio
+import inspect
+import uuid
+from typing import Any, Dict, List, Optional, Set, Union
+
+import anyio
+import httpx
+from typing_extensions import Literal
+
+from prefect.client.orchestration import get_client
+from prefect.context import FlowRunContext
+from prefect.flows import Flow
+from prefect.logging import get_logger
+from prefect.settings import (
+    PREFECT_RUNNER_PROCESS_LIMIT,
+    PREFECT_RUNNER_SERVER_HOST,
+    PREFECT_RUNNER_SERVER_PORT,
+)
+from prefect.states import Pending
+from prefect.tasks import Task
+from prefect.utilities.asyncutils import sync_compatible
+
+logger = get_logger("webserver")
+
+
+async def get_current_run_count() -> int:
+    async with httpx.AsyncClient() as client:
+        response = await client.get(
+            f"http://{PREFECT_RUNNER_SERVER_HOST.value()}"
+            f":{PREFECT_RUNNER_SERVER_PORT.value()}/run_count"
+        )
+        response.raise_for_status()
+        return response.json()
+
+
+async def _submit_flow_to_runner(
+    flow: Flow,
+    parameters: Dict[str, Any],
+    # capture_errors: bool = SETTING.value()?
+) -> uuid.UUID:
+    """
+    Run a callable in the background via the runner webserver.
+
+    Args:
+        prefect_callable: the callable to run, e.g. a flow or task
+        parameters: the keyword arguments to pass to the callable
+        timeout: the maximum time to wait for the callable to finish
+        poll_interval: the interval (in seconds) to wait between polling the callable
+    """
+    from prefect.engine import (
+        _dynamic_key_for_task_run,
+        collect_task_run_inputs,
+        resolve_inputs,
+    )
+
+    async with get_client() as client:
+        parent_flow_run_context = FlowRunContext.get()
+
+        task_inputs = {
+            k: await collect_task_run_inputs(v) for k, v in parameters.items()
+        }
+
+        dummy_task = Task(name=flow.name, fn=flow.fn, version=flow.version)
+        parent_task_run = await client.create_task_run(
+            task=dummy_task,
+            flow_run_id=parent_flow_run_context.flow_run.id,
+            dynamic_key=_dynamic_key_for_task_run(parent_flow_run_context, dummy_task),
+            task_inputs=task_inputs,
+            state=Pending(),
+        )
+
+        parameters = await resolve_inputs(parameters)
```

Comment:
Note for later - I think we should move this up earlier in this code block so that if there's an exception resolving inputs we haven't already created the task run.

Reply:
ahhh yes good call

Reply:
not totally sure why though